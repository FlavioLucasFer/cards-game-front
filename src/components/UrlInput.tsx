import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    label: PropTypes.string,
    labelClassname: PropTypes.string,
    urlPrefix: PropTypes.string.isRequired,
};

export type UrlInputProps = React.HTMLProps<HTMLInputElement> & PropTypes.InferProps<typeof propTypes>;

const UrlInput: React.FC<UrlInputProps> = props => {
    const {
        label,
        labelClassname,
        urlPrefix,
        id = '',
        ...inputProps
    } = props;

    return (
        <div>
            {label &&
                <label 
                    htmlFor={id} 
                    className={`form-label ${labelClassname || ''}`}
                >
                    {label}
                </label>
            }
            <div className='input-group mb-3'>
                <span 
                    id={`${id}-url-prefix`}
                    className='input-group-text' 
                >
                    {urlPrefix}
                </span>
                <input 
                    id={id} 
                    className={`form-control ${inputProps?.className || ''}`} 
                    aria-describedby={`${id}-url-prefix`} 
                    {...inputProps}
                />
            </div>
        </div>
    );
}

UrlInput.propTypes = propTypes;

export default UrlInput;
