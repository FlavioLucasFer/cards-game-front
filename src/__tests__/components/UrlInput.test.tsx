import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import UrlInput from '../../components/UrlInput';

configure({ adapter: new Adapter() });

describe('UrlInput tests', () => {
    it('should render properly', () => {
        const wrapper = shallow(
            <UrlInput 
                urlPrefix='http://localhost:3000/'
            />
        );

        expect(wrapper.debug()).toMatchSnapshot();
    });
});
