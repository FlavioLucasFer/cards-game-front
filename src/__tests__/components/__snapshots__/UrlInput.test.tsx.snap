// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInput tests should render properly 1`] = `
"<div>
  <div className=\\"input-group mb-3\\">
    <span id=\\"-url-prefix\\" className=\\"input-group-text\\">
      http://localhost:3000/
    </span>
    <input id=\\"\\" className=\\"form-control \\" aria-describedby=\\"-url-prefix\\" />
  </div>
</div>"
`;
